test.nix

```diff
@@ -121,6 +121,12 @@ let
         cargoOptions = (opts: opts ++ [ "--locked" ]);
       };
 
+      git-dep-with-submodule = naersk.buildPackage {
+        doCheck = true;
+        src = ./test/git-dep-with-submodule;
```

Comment:
I think you forgot to check in the testcase itself.

Reply:
Yup I did! 🤦‍♀️ 

Good news  is this new commit includes a fix!